pkg/sql/lease.go

```diff
@@ -791,9 +817,10 @@ func (t *tableState) removeLease(lease *LeaseState, m *LeaseManager) {
 	}
 
 	// Release to the store asynchronously, without the tableState lock.
-	if err := t.stopper.RunAsyncTask(ctx, func(ctx context.Context) {
-		m.LeaseStore.Release(ctx, t.stopper, lease)
-	}); err != nil {
+	if err := t.stopper.RunLimitedAsyncTask(ctx, removeLeaseSem, true,
```

Comment:
I worry about changing this to use the RunLimitedAsyncTask() because it blocks while holding on to the lock over all leases for a table. I think 1.0 users are likely to use very few tables and so it's very likely there will be only a few of these async tasks created.